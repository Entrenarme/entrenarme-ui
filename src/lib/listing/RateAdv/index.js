// @flow
import * as React from 'react';
import styled from 'styled-components';

import colors from '../../helpers/colors';
import P from '../../core/Text/Paragraph';
import Button from '../../core/Button';

const Wrapper = styled.div`
  width: 130px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

type ContainerProps = {
  options: { appendButtonText: ?string },
};

const Container = styled.div`
  background-color: ${colors.gray.light};
  width: 130px;
  height: 70px;
  border-radius: 4px;
  ${(props: ContainerProps) =>
    props.options && props.options.appendButtonText
      ? `
      border-bottom-left-radius: 0;
      border-bottom-right-radius: 0;
    `
      : ''};
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  z-index: 2;
`;

type Props = {
  topText: React.Node,
  bottomText: React.Node,
  amount: number,
  currency: string,
  appendButtonText: ?string,
};

const Price = styled.div`
  display: flex;
  justify-content: center;
  align-items: baseline;
`;

const RatesAdv = ({
  topText,
  bottomText,
  amount,
  currency,
  appendButtonText,
}: Props) => (
  <Wrapper>
    <Container options={{ appendButtonText }}>
      <P size="xs" color={colors.primary.default} textAlign="center" noMargin>
        {topText}
      </P>
      <Price>
        <P
          size="large"
          color={colors.primary.default}
          font="main"
          textAlign="center"
          noMargin
        >
          {amount}
        </P>
        <P
          size="medium"
          color={colors.primary.default}
          font="main"
          textAlign="center"
          noMargin
        >
          {currency}
        </P>
      </Price>
      <P size="xs" color={colors.primary.default} textAlign="center" noMargin>
        {bottomText}
      </P>
    </Container>
    {appendButtonText && (
      <Button fullWidth size="xs" style={{ marginTop: '-2px' }}>
        {appendButtonText}
      </Button>
    )}
  </Wrapper>
);

RatesAdv.defaultProps = {
  appendButtonText: null,
};

export default RatesAdv;
