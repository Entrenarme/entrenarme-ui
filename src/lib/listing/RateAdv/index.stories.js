// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';

import RatesAdv from './index';

storiesOf('RatesAdv', module).add('default', () => (
  <RatesAdv
    currency="€"
    amount={12}
    topText="desde"
    bottomText="persona/sesión"
  />
));
