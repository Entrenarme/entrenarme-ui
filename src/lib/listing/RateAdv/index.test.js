import * as React from 'react';
import { render } from 'react-testing-library';

import RateAdv from './index';

test('renders button if appendButtonText', () => {
  const { getByTestId } = render(<RateAdv appendButtonText="button text" />);

  const button = getByTestId('appended-button');

  expect(button).toBeTruthy();
  expect(button.nodeName).toBe('BUTTON');
});

test('dont renders button if NO appendButtonText', () => {
  const { queryByTestId } = render(<RateAdv />);

  const button = queryByTestId('appended-button');

  expect(button).toBeNull();
});

test('renders the expected props', () => {
  const { container } = render(
    <RateAdv
      currency="€"
      amount={12}
      topText="desde"
      bottomText="persona/sesión"
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});
