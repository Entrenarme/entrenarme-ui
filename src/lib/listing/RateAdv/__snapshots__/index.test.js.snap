// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the expected props 1`] = `
<div
  class="sc-bxivhb bQXWIa"
>
  <div
    class="sc-ifAKCX fqXEi"
  >
    <p
      class="sc-bdVaJa gIQiiZ"
    >
      desde
    </p>
    <div
      class="sc-EHOje kuAuJM"
    >
      <p
        class="sc-bdVaJa etpXmm"
      >
        12
      </p>
      <p
        class="sc-bdVaJa gGAOdY"
      >
        €
      </p>
    </div>
    <p
      class="sc-bdVaJa gIQiiZ"
    >
      persona/sesión
    </p>
  </div>
</div>
`;
